core/impl/src/main/java/com/blazebit/persistence/impl/EntityMetamodelImpl.java

```diff
@@ -337,6 +339,20 @@ private TemporaryExtendedManagedType collectColumnNames(EntityType<?> e, Map<Str
                         }
                     }
                 }
+                else if (jpaProvider.getJpaMetamodelAccessor().isCompositeUserType(delegate, attribute)) {
+                    Collection<String> compositePropertyNames = jpaProvider.getJpaMetamodelAccessor().getCompositePropertyNames(delegate, attribute);
+
+
+
+                    for (String compositePropertyName : compositePropertyNames) {
+                        String idPath = attributeName + "." + compositePropertyName;
+                        List<Attribute<?,?>> idParents = new ArrayList<>();
+                        idParents.add(attribute);
+                        AttributeEntry attributeEntry = new AttributeEntry(jpaProvider, type, attribute, idPath, BigDecimal.class, idParents, null);
```

Comment:
We need a way to access the property information like the java type, column type, nullability etc. Here you are always registering `BigDecimal` as type. I know it's just a PoC, I'm just mentioning it.